File: CS-Examples/14_Hyperlinks/WriteHyperlinks.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace WriteHyperlinks
{

	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing workbook from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WriteHyperlinks.xlsx");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set the text for cell B9 as "Home page"
            sheet.Range["B9"].Text = "Home page";

            // Add a hyperlink to cell B10
            HyperLink hylink1 = sheet.HyperLinks.Add(sheet.Range["B10"]);
            hylink1.Type = HyperLinkType.Url;
            hylink1.Address = @"http://www.e-iceblue.com";

            // Set the text for cell B11 as "Support"
            sheet.Range["B11"].Text = "Support";

            // Add a hyperlink to cell B12
            HyperLink hylink2 = sheet.HyperLinks.Add(sheet.Range["B12"]);
            hylink2.Type = HyperLinkType.Url;
            hylink2.Address = "mailto:support@e-iceblue.com";

            // Set the text for cell B13 as "Forum"
            sheet.Range["B13"].Text = "Forum";

            // Add a hyperlink to cell B14
            HyperLink hylink3 = sheet.HyperLinks.Add(sheet.Range["B14"]);
            hylink3.Type = HyperLinkType.Url;
            hylink3.Address = "https://www.e-iceblue.com/forum/";

            // Specify the output file name for the modified workbook
            String result = "Output_WriteHyperlinks.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(result);
		}

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

	}
}

File: CS-Examples/14_Hyperlinks/WriteHyperlinks.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace WriteHyperlinks

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Load an existing workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WriteHyperlinks.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the text of cell B9 as "Home page"
            sheet.Range("B9").Text = "Home page"

            ' Create a hyperlink object and associate it with cell B10
            Dim hylink1 As HyperLink = sheet.HyperLinks.Add(sheet.Range("B10"))

            ' Set the type of the hyperlink to a URL
            hylink1.Type = HyperLinkType.Url

            ' Set the address of the hyperlink to "http://www.e-iceblue.com"
            hylink1.Address = "http://www.e-iceblue.com"

            ' Set the text of cell B11 as "Support"
            sheet.Range("B11").Text = "Support"

            ' Create a hyperlink object and associate it with cell B12
            Dim hylink2 As HyperLink = sheet.HyperLinks.Add(sheet.Range("B12"))

            ' Set the type of the hyperlink to a URL
            hylink2.Type = HyperLinkType.Url

            ' Set the address of the hyperlink to "mailto:support@e-iceblue.com"
            hylink2.Address = "mailto:support@e-iceblue.com"

            ' Set the text of cell B13 as "Forum"
            sheet.Range("B13").Text = "Forum"

            ' Create a hyperlink object and associate it with cell B14
            Dim hylink3 As HyperLink = sheet.HyperLinks.Add(sheet.Range("B14"))

            ' Set the type of the hyperlink to a URL
            hylink3.Type = HyperLinkType.Url

            ' Set the address of the hyperlink to "https://www.e-iceblue.com/forum/"
            hylink3.Address = "https://www.e-iceblue.com/forum/"

            ' Define the output file name as "Output_WriteHyperlinks.xlsx"
            Dim result As String = "Output_WriteHyperlinks.xlsx"

            ' Save the modified workbook to the specified file path using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
